typescript/src/schema/resourceDatabase.ts

```diff
@@ -55,32 +55,47 @@ export class ResourceDatabase {
       }
       return;
     }
-    const name = arr[1];
-
-    const pattern = resource.pattern;
-    if (!pattern?.[0]) {
+    const patterns = resource.pattern;
+    if (!patterns?.[0]) {
       if (errorLocation) {
         console.warn(
           `Warning: ${errorLocation} refers to a resource which does not have a proper pattern: ${resource}`
         );
       }
       return;
     }
-    const params = pattern[0].match(/{[a-zA-Z_]+(?:=.*?)?}/g) || [];
-    for (let i = 0; i < params.length; i++) {
-      params[i] = params[i].replace(/{([a-zA-Z_]+).*/, '$1');
-    }
-
-    const resourceDescriptor: ResourceDescriptor = Object.assign(
-      {
+    const multiPattern = patterns!.length > 1;
+    // only one pattern exists for the resource.
+    if (!multiPattern) {
+      const name = arr![1];
+      const params = this.getParams(patterns![0]);
+      const resourceDescriptor = this.getResourceDescriptor(
         name,
         params,
-      },
-      resource
-    );
-
-    this.patterns[pattern?.[0]] = resourceDescriptor;
-    this.types[resourceDescriptor.type!] = resourceDescriptor;
+        resource
+      );
+      this.patterns[patterns?.[0]] = resourceDescriptor;
+      this.types[resourceDescriptor.type!] = resourceDescriptor;
+    }
+    // resource: {name, type, pattern: [p1, p2]}
+    // register resource does: in type map {type: { name, type, pattern: [p1, p2]} }
+    //                         in pattern map {p1: { name1, type, p1} , p2: { name2, type, p2}}
+    else {
+      for (const pattern of patterns!) {
+        const params = this.getParams(pattern);
+        const name = params.map(r => r.toPascalCase()).join('');
```

Comment:
I don't think we should pascal case the names here (PascalCase for resource names is kind of TypeScript specific and should go to templates). Can we leave the names in snake case (as they come in protos) and use pascal case only in templates?

Reply:
yes sure, so here for the pattern `organizations/{organization}/inspectTemplates/{inspect_template}` it will combine the two params into `OrganizationInspectTemplates` in pascal case, but I think `organization_inspect_template` and convert the name to pascal case in template should work. Thanks